pandas/core/ops/__init__.py

```diff
@@ -1038,8 +1038,14 @@ def wrapper(self, other, axis=None):
             # Defer to DataFrame implementation; fail early
             return NotImplemented
 
-        elif isinstance(other, ABCSeries) and not self._indexed_same(other):
+        if isinstance(other, ABCSeries) and not self._indexed_same(other):
             raise ValueError("Can only compare identically-labeled Series objects")
+        elif (
+            is_list_like(other)
+            and len(other) != len(self)
+            and not isinstance(other, (set, frozenset))
```

Comment:
@jbrockmendel maybe by accident due to several PRs that were related / rebase after each other, but this seemingly introduced again what you removed in https://github.com/pandas-dev/pandas/pull/27873 (although the other PR was merged later ..)

No matter how it happened ;), the check is still in master ..

Reply:
yah this looks like a rebase screwup, will address